### DISCUSSION TITLE
MaxInstances an visibility errors

### DISCUSSION THREAD

**Discussion by narmuzz** [webdriverio/webdriverio#11898] [Q&A] [answered]
Hi! I've been having some issues with the .isDisplayed(), waitForDisplayed(), and browser.isElementDisplayed() methods with them either seeing a visible element as hidden or viceversa. The .click() and browser.elementClick() methods are also failing every now and then.

I've noticed that these errors became way more common as I increased the maxInstances value. While running with maxInstances = 4, I saw almost no error, around one every 50 tests. When running with maxInstances = 10, I had about 10 or 12 such errors every 24 tests. 

I've noticed that different machines have a different "breaking point" and also that said point is lower (i.e. less maxInstances) if the tests are running in a dockerized enviroment (my guess here is that docker's resource drain makes the situation worse).

I'm not so much looking for a solution as I'm to looking to understand what's happening. Is it a known thing that increasing maxInstances beyond a certain point will lead to a marked increase in errors, or is this weird thing on my end, and if so, is there any recomendation as to what value to use?

Thanks!

**Comment by christian-bromann** [comment, marked as answer]
Hey @narmuzz , thanks for creating this discussion. My hunch here is that by adding more browser to your pipeline it results in more memory being consumed causing race conditions to appear and generally things to run slower. I don't have a simple solution to that unfortunately.

**Reply by narmuzz** [reply]
I was thinking something along the same lines. Knowing this is not just me being ignorant will really help in deciding how to manage our tests. Thank you very much!

**Reply by christian-bromann** [reply]
I am not sure how you run your grid but I had the best experience so far using cloud vendors as they can give you browser in the cloud that barely behave differently. Now if you run all your tests on a single CI machine, e.g. having 100 or more parallel sessions using a cloud vendor, things can become unreliable as well due to a lot of I/O happening. WebdriverIO has support for sharding so you can split up that load on multiple CI machines too to get more stable results.
